File: comms/bulk-messaging/sender-pools/list-sender-pools/list-sender-pools.py
from twilio_comms import TwilioComms
import datetime

client = TwilioComms(
    account_id="TWILIO_ACCOUNT_SID",
    auth_token="TWILIO_AUTH_TOKEN",
)

client.sender_pools.list(
    start_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    end_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    operation_id="comms_operation_01h9krwprkeee8fzqspvwy6nq8",
    page_token="pageToken",
    page_size=50,
)

File: comms/bulk-messaging/sender-pools/list-sender-pools/list-sender-pools.cs
using Twilio.Comms;
using System.Threading.Tasks;
using System;
using System.Globalization;

public partial class Examples
{
    public async Task Example() {
        var client = new TwilioComms(
            accountId: "TWILIO_ACCOUNT_SID",
            authToken: "TWILIO_AUTH_TOKEN"
        );

        await client.SenderPools.ListAsync(
            new SenderPoolsListRequest {
                StartDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                EndDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                OperationId = "comms_operation_01h9krwprkeee8fzqspvwy6nq8",
                PageToken = "pageToken",
                PageSize = 50
            }
        );
    }

}
